proxy/src/main/java/com/wavefront/agent/logsharvesting/FlushProcessorContext.java

```diff
@@ -13,25 +14,43 @@
   private PointHandler pointHandler;
   private String prefix;
 
-  public FlushProcessorContext(TimeSeries timeSeries, String prefix, PointHandler pointHandler) {
+  FlushProcessorContext(TimeSeries timeSeries, String prefix, PointHandler pointHandler) {
     this.timeSeries = TimeSeries.newBuilder(timeSeries).build();
     this.pointHandler = pointHandler;
     this.prefix = prefix;
   }
 
-  public TimeSeries getTimeSeries() {
-    return timeSeries;
+  String getMetricName() {
+    return timeSeries.getMetric();
   }
 
-  public PointHandler getPointHandler() {
-    return pointHandler;
-  }
-
-  public ReportPoint.Builder reportPointBuilder() {
+  private ReportPoint.Builder reportPointBuilder() {
     return ReportPoint.newBuilder()
         .setHost(timeSeries.getHost())
         .setAnnotations(timeSeries.getAnnotations())
         .setMetric(prefix == null ? timeSeries.getMetric() : prefix + "." + timeSeries.getMetric());
   }
 
+  void report(double value) {
+    pointHandler.reportPoint(
+        reportPointBuilder().setValue(value).setTimestamp(System.currentTimeMillis()).build(), null);
```

Comment:
i would suggest using a single timestamp for the entire reporting context so that at least one can correlate them as a single reporting cycle on the server